code/game/gamemodes/miniantags/demons/slaughter demon/newSlaughter.dm

```diff
@@ -0,0 +1,284 @@
+		REMOVE_TRAIT(M, TRAIT_UNREVIVABLE, "demon")
+		release_consumed(M)
+	. = ..()
+
+/mob/living/simple_animal/demon/slaughter_demon/proc/release_consumed(mob/living/M)
+	M.forceMove(get_turf(src))
+
+/mob/living/simple_animal/demon/slaughter_demon/proc/perform_phasein()
+	var/foundtarget = FALSE
+	var/turf/origin_turf = channel_target
+	var/turf/tele_turf = channel_target
+	for(var/obj/effect/decal/cleanable/C in tele_turf)
+		if(C.can_bloodcrawl_in())
+			foundtarget = TRUE
+			break
+	var/foundblood = FALSE
+	if(!foundtarget)
+		for(var/obj/effect/decal/cleanable/C in view(2, tele_turf))
+			if(C.can_bloodcrawl_in())
+				if(!foundblood)
+					foundblood = TRUE
+					tele_turf = get_turf(C)
+					break
+	if(!foundblood && !foundtarget)
+		channeling = FALSE
+		to_chat(src, "The blood has all been cleaned!")
+		REMOVE_TRAIT(src, TRAIT_IMMOBILIZED, "channelingblood")
+		return
+	for(var/obj/effect/decal/cleanable/C in view(2, origin_turf))
+		if(C.can_bloodcrawl_in())
+			for(var/mob/living/M in get_turf(C))
+				if(ishuman(M))
+					var/mob/living/carbon/human/H = M
+					H.bleed_rate += 10
+				M.Weaken(5 SECONDS)
+				new /obj/effect/temp_visual/cult/sparks(get_turf(M))
+	src.forceMove(tele_turf)
+	src.client.eye = src
+	phased = TRUE
+	channeling = FALSE
+	REMOVE_TRAIT(src, TRAIT_IMMOBILIZED, "channelingblood")
+	icon_state = "daemon"
+	new /obj/effect/temp_visual/dir_setting/bloodcrawl(tele_turf, src.dir, "jaunt")
+	if(prob(25))
+		var/list/voice = list('sound/hallucinations/behind_you1.ogg', 'sound/hallucinations/im_here1.ogg', 'sound/hallucinations/turn_around1.ogg', 'sound/hallucinations/i_see_you1.ogg')
+		playsound(tele_turf, pick(voice), 50, 1, -1)
+	channel_target.visible_message("<span class='warning'><b>[src] rises out of [channel_target]!</b>")
+	playsound(tele_turf, 'sound/misc/exit_blood.ogg', 100, 1, -1)
+	bloodspots = list()
+
+/mob/living/simple_animal/demon/slaughter_demon/proc/absorb()
+	for(var/mob/living/L in range(1, get_turf(src)))
+		if(L.stat != DEAD)
+			continue
+		devoured++
+		L.forceMove(src)
+		ADD_TRAIT(L, TRAIT_UNREVIVABLE, "demon")
+		consumed_mobs.Add(L)
+		if(ishuman(L))
+			var/mob/living/carbon/human/H = L
+			if(H.w_uniform && istype(H.w_uniform, /obj/item/clothing/under))
+				var/obj/item/clothing/under/U = H.w_uniform
+				U.sensor_mode = SENSOR_OFF
+		if(L.mind)
+			if(!charge_holder.bloodcharge || !slam_holder.bloodcharge)
+				charge_holder.bloodcharge = TRUE
+				slam_holder.bloodcharge = TRUE
+			mindsabsorbed++
+			maxHealth += 10
+			adjustHealth(-100)
+			L.forceMove(src)
+	while(mindsabsorbed % 4 == 0 && mindsabsorbed > 0)
+		mindsabsorbed -= 4
+		maxHealth += 50
+
+/mob/living/simple_animal/demon/slaughter_demon/proc/perform_phaseout()
+	icon_state = "daemon"
+	var/obj/effect/decal/foundblood = 0
+	for(var/obj/effect/decal/cleanable/C in channel_target)
+		if(C.can_bloodcrawl_in())
+			foundblood = C
+			break
+	if(!foundblood)
+		to_chat(src, "The target blood has been cleaned!")
```

Comment:
```suggestion
		to_chat(src, "<span class='warning'>The blood you were channeling has been cleaned away!<span>")
```
this might be for debugging? but its probably a good idea to have some feedback on this